backend/controllers/locationZoneController.js

```diff
@@ -0,0 +1,165 @@
+
+  res.status(200).json({
+    success: true,
+    payload: {}
+  })
+});
+
+// @desc    Update a location zone name or description
+// @route   PUT /api/v1/locations/zones/:zoneId
+// @access  Public
+exports.updateZoneNameDescription = asyncHandler(async (req, res, next) => {
+  const zone = await LocationZone.findByIdAndUpdate(req.params.zoneId, req.body, {
+    new: true,
+    runValidators: true
+  });
+
+  res.status(200).json({
+    success: true,
+    payload: zone
+  })
+});
+
+function getPolyginCentroidAvg(path) {
+  // get the centroid of a polygon by taking the average of x, and the average of y
+
+  let lngAverage = 0;
+  let latAverage = 0;
+
+  path.forEach(vertex => {
+    lngAverage += vertex.lng;
+    latAverage += vertex.lat;
+  })
+
+  lngAverage /= path.length;
+  latAverage /= path.length;
+
+  const center = {
+    lat: latAverage,
+    lng: lngAverage
+  }
+
+  return center;
+}
+
+function getPolygonCentroidSum(path) {
+  // get the centroid of a polygon by using the formula in http://paulbourke.net/geometry/polygonmesh/centroid.pdf
+
+  /*
+    Calculating the center of mass of the outlined zone:
+
+    Area of a polygon is defined as the following:
+        A = 1/2 * SUM(x(i)y(y+1) - x(i+1)y(i)) for i=0 TO N-1
+    
+    X coordinate of the center is defined as the following:
+        Cx = 1/(6A) * SUM((x(i) + x(i+1))*(x(i)y(i+1) - x(i+1)y(i))) for i=0 TO N-1
+    Y coordinate of the center is defined as the following:
+        Cy = 1/(6A) * SUM((y(i) + y(i+1))*(x(i)y(i+1) - x(i+1)y(i))) for i=0 TO N-1
+
+    Y = latitude
+    X = longitude
+    */
+
+  // get the first point in the path and add it as the last point
+  path.push(path[0]);
+
+  // calculate the area
+  let zoneArea = 0
+  let currentSum = 0;
+  for (let i = 0; i < path.length - 1; i++) {
+    let x = path[i].lng;
+    let xPlusOne = path[i + 1].lng;
+    let y = path[i].lat;
+    let yPlusOne = path[i + 1].lat;
+
+    currentSum = (x * yPlusOne) - (xPlusOne * y);
+    zoneArea += currentSum;
+  }
+
+  zoneArea = zoneArea / 2;
+
+  // calculate the center X coordinate
+  let xCenter = 0;
+  currentSum = 0;
+  for (let i = 0; i < path.length - 1; i++) {
+    let x = path[i].lng;
+    let xPlusOne = path[i + 1].lng;
+    let y = path[i].lat;
+    let yPlusOne = path[i + 1].lat;
+
+    currentSum = (x + xPlusOne) * ((x * yPlusOne) - (xPlusOne * y));
+    xCenter += currentSum;
+  }
+
+  xCenter = xCenter / (6 * zoneArea);
+
+  // calculate the center Y coordinate
+  let yCenter = 0;
+  currentSum = 0;
+  for (let i = 0; i < path.length - 1; i++) {
+    let x = path[i].lng;
+    let xPlusOne = path[i + 1].lng;
+    let y = path[i].lat;
+    let yPlusOne = path[i + 1].lat;
+
+    currentSum = (y + yPlusOne) * ((x * yPlusOne) - (xPlusOne * y));
+    yCenter += currentSum;
+  }
+
+  yCenter = yCenter / (6 * zoneArea);
+
+  // create the center object
+  const center = {
+    lat: yCenter,
+    lng: xCenter
+  }
+
+  return center;
+}
```

Comment:
Remove this dead code because I cannot find a reference to it.

Reply:
Addressed in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/87/commits/b3016ee9df8271d80dc6fc49195c95de810e1398)